runtime/doc/dev_style.txt

```diff
@@ -859,6 +859,9 @@ execute, simply `assert`: >c
         assert(false);
     }
 
+Switch statements that are conditional on an enumerated value may not have a
+`default` case if it is exhaustive.
```

Comment:
line 846 already says "If not conditional on an enumerated value". Do we need this extra note? The compiler enforces the "exhaustive" requirement, right?

Reply:
There was clearly confusion regarding this as we have had 2-3 discussions about this topic in Matrix. If the ambiguity causes confusion, I think it's best to just have it stated explicitly.

Reply:
The problem is that "the compiler" (the minimum compiler versions as used on CI) has **never** done this correctly so far in the lifetime of the nvim project, which means that in practice we have fallen through (no pun intended) to the next sentence also  for enum-based switches and added `default: abort()` to appease these compiler versions. 

Upgrading all the CI compilers will enable us to finally perform the original intention of this paragraph. Doing an explicit mention now is a workaround to having the style guide be dependent of "the compiler" in the past (which is _not_ how the style guide should have been written.. the style guide should document what _we_ want, not make references to what some unspecified compiler version wants)